### DISCUSSION TITLE
Question about "generic API appliance" and "Washing machine appliance"

### DISCUSSION THREAD

**Discussion by rudizl** [astrandb/miele#180] [Q&A] [answered]
Hi @astrandb, 
compraring the "Generic API" and "Wahing machine API" I found, that state `signalDoor` is only in the first one, but I'm still able to receive that info for my WCR860. Is it possible, that state `light` and eventually action `light` can be implemented for washing machines with light too? I've tried to add WASHING_MACHINE in light.py, but without succes, so I hope I'm missing someting. Any ideas?

**Comment by astrandb** [comment]
The public API does only expose a limited set of data depending on the model and version of the appliance. Washing machine light is not listed as supported in the API spec.
You can take a look in the diagnostic download for your washing machine and find out what actual data is reported and what actions that are supported. If you need help with interpretation you can upload the diagnostic file here.

**Comment by rudizl** [comment]
I will need help for sure :)
Here is the file:
[miele-675dbf2ebf5ffba12fe4ec1d5604fa1c-Washing machine-f5fa1462007bbb5f74ea1c26626380d7.json.txt](https://github.com/astrandb/miele/files/10219586/miele-675dbf2ebf5ffba12fe4ec1d5604fa1c-Washing.machine-f5fa1462007bbb5f74ea1c26626380d7.json.txt)

**Comment by astrandb** [comment, marked as answer]
`"state|light": null,` indicates that there is no API support for light status for this device.

```json
    "actions": {
      "processAction": [],
      "light": [],
      "ambientLight": [],
```
And no action to control light.
😞 

I interpret the generic API as a sum of what is supported. The machine specific API describes what features that are available for that specific device type. Then all (older) washing machines do not support all features.

**Reply by rudizl** [reply]
Thank you for your quick responce. No "Licht, mehr Licht" for me, bad Miele, bad Miele ;)
